File: Java/203_Remove_Linked_List_Elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

// Time O(n) Space(1)

class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode sen = new ListNode(0);
        sen.next = head;
        
        ListNode prev = sen;
        ListNode curr = head;
        while(curr != null){
            if(curr.val == val) prev.next = curr.next;
            else prev = curr;
            curr = curr.next;
        }
        return sen.next;
        
    }
}

File: Python/203_remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy_node = ListNode(0)
        dummy_node.next = head

        prev, cur = dummy_node, head
        while cur is not None:
            if cur.val == val:
                prev.next= cur.next
            else:
                prev= cur
            cur = cur.next
        return dummy_node.next
